RabbitMQ.Stream.Client/StreamSystem.cs

```diff
@@ -120,7 +120,48 @@ private static void CheckLeader(StreamInfo metaStreamInfo)
             }
         }
 
-        public async Task<Producer> CreateProducer(ProducerConfig producerConfig)
+        public async Task<IProducer> CreateSuperStreamProducer(SuperStreamProducerConfig superStreamProducerConfig)
+        {
+            await MayBeReconnectLocator();
+            if (superStreamProducerConfig.SuperStream == "")
+            {
+                throw new CreateProducerException($"Super Stream name can't be empty");
+            }
+
+            if (superStreamProducerConfig.MessagesBufferSize < Consts.MinBatchSize)
+            {
+                throw new CreateProducerException(
+                    $"Batch Size must be bigger than 0");
+            }
+
+            if (superStreamProducerConfig.Routing == null)
+            {
+                throw new CreateProducerException(
+                    $"Routing Key Extractor must be provided");
+            }
```

Comment:
Would make sense to make this validation in the `SuperStreamProducerConfig` constructor?
The rationale is to not let the user create an invalid `SuperStreamProducerConfig` that we know is not going to work; instead of letting create an invalid config and then failing to create the producer.

Reply:
yes, good point. 
We should do the same to the standard producer

Reply:
@Zerpet, this would introduce a breaking change. The `SuperStreamProducerConfig`  would be:
`new SuperStreamProducerConfig(streamName)` instead of `new SuperStreamProducerConfig()`

So the `stream name` would be mandatory

Reply:
Agreed with Gabriele to address this conversation in a follow up issue.